### PR TITLE
Features/retrieve_last_ingested

### PR DESCRIPTION
**Pull request by noctillion**
This Pull Request includes changes related to the retrieval of ingestion time for variants:

CreatedTime Metadata: With the updates in this PR, during the variant ingestion process, is captured and stored the CreatedTime for each variant. This metadata provides insights into when each variant was added.

Latest Created Field in Variants Overview: A new field named latest_created has been added to the variants overview. This field indicates the most recent creation timestamp among all variants.

Last Ingested File Information: For datasets, it is included information about the last ingested file. These changes in the GetDatasetDataTypes function provide each dataset's most recent updates.

### CODE REVIEW

**Comment by noctillion** [comment]
This Pull Request include changes related to the retrieval of ingestion time for variants:

CreatedTime Metadata: With the updates in this PR, during the variant ingestion process, is captured and stored the CreatedTime for each variant. This metadata provides insights into when each variant was added.

Latest Created Field in Variants Overview: A new field named latest_created has been added to the variants overview. This field indicates the most recent creation timestamp among all variants.

Last Ingested File Information: For every dataset, we now include information about the last ingested file. These changes in the GetDatasetDataTypes function provide each dataset's most recent data updates.

File: src/api/mvc/data-types/main.go [hunk 1]
```diff
@@ -19,26 +19,68 @@ var variantDataTypeJson = map[string]interface{}{
...
 		})
 	}
 
-	variantDataTypeJson["count"] = sumAllValues(resultsMap["sampleIDs"])
+	return c.JSON(http.StatusOK, []map[string]interface{}{
+		variantData,
+	})
+}
+
+func GetReducedDataTypes(c echo.Context) error {
```
Comment (v-rocheleau) [inline src/api/mvc/data-types/main.go:53]: Can you add a comment describing what this function does? How is it different from GetDataTypes?

**Review by v-rocheleau — CHANGES_REQUESTED** [review]
Error handling stuff, also should fix tests

File: src/api/repositories/elasticsearch/variants.go [hunk 2]
```diff
@@ -322,6 +322,83 @@ func GetDocumentsContainerVariantOrSampleIdInPositionRange(cfg *models.Config, e
...
+	if hits, found := result["hits"].(map[string]interface{}); found {
+		if hitSlice, hitFound := hits["hits"].([]interface{}); hitFound && len(hitSlice) > 0 {
+			if firstHit, firstHitFound := hitSlice[0].(map[string]interface{}); firstHitFound {
+				if source, sourceFound := firstHit["_source"].(map[string]interface{}); sourceFound {
+					if created, createdFound := source["createdTime"].(string); createdFound {
+						parsedTime, err := time.Parse(time.RFC3339, created)
+						if err == nil {
+							mostRecentTimestamp = parsedTime
+						} else {
+							fmt.Printf("Error parsing 'createdTime' timestamp: %s\n", err)
```
[the remark this answers is not in the document]
Comment (v-rocheleau) [reply src/api/repositories/elasticsearch/variants.go:389]: The final return will also be an epoch zero timestamp

File: src/api/services/variants/main.go [hunk 3]
```diff
@@ -29,6 +30,32 @@ func GetVariantsOverview(es *elasticsearch.Client, cfg *models.Config) (map[stri
 	resultsMux := sync.RWMutex{}
 
 	var wg sync.WaitGroup
+
+	callProcessLatestCreated := func(key string, keyword string, _wg *sync.WaitGroup) {
```
Comment (v-rocheleau) [inline src/api/services/variants/main.go:34]: Can you add comments in this function?

**Review by v-rocheleau — CHANGES_REQUESTED** [review]

**Review by v-rocheleau — APPROVED** [review]

**Pull request merged**